src/dvc_objects/db.py

```diff
@@ -129,7 +130,7 @@ def add(
         if self.read_only:
             raise ObjectDBPermissionError("Cannot add to read-only ODB")
 
-        if self.exists(oid):
+        if check_exists and self.exists(oid):
```

Comment:
This exists check is causing a giant performance hit in regular DVC remote operations. In dvc-data `hashfile.transfer`, we already check status, and are explicitly only adding new objects to the odb. But this call is still being done for every object added to the odb. So we are essentially checking status twice for every object we tried to push

Reply:
One of the problems is that our indexes that we computed for transfer are not used at all in the db. Ideally this `exists` call should completely go away or at the very least shortcut through the index.

Reply:
Yeah I'd like to remove it eventually, but I think there's some places in dvc-data index where we are expecting odb.add to do the exists check for us